src/components/RadioInput/index.tsx

```diff
@@ -10,13 +10,10 @@ export const RadioInput: React.FC<{
   onChange?: (event: RadioInputEvent) => void;
 }> = (props) => {
   const { children, checked = false, onChange = () => null, ...rest } = props;
-  const [state, setState] = useState(checked);
```

Comment:
This allows the component to maintain it's own state. Why do you think removing it is a better approach?